src/main/java/com/andreamazzarella/contact_manager/InMemoryRepository.java

```diff
@@ -0,0 +1,28 @@
+package com.andreamazzarella.contact_manager;
+
+import java.util.ArrayList;
+
+public class InMemoryRepository {
+    private ArrayList<Contact> contacts;
+
+    public InMemoryRepository() {
+        this.contacts = new ArrayList<>();
+    }
+
+    public ArrayList allContacts() {
+        return this.contacts;
+    }
+
+    public void addContact(Contact contact) {
+        this.contacts.add(contact);
+    }
+
+    public ArrayList findContact(final String searchTerm) {
+        ArrayList<Contact> searchResults = new ArrayList<>();
+        for (Contact contact : this.contacts) {
```

Comment:
I would look into the Java 8 streams API - this could something like `this.contacts.stream.filter(c => c.nameContains(searchTerm)`